### ISSUE TITLE
Container fails not enough memory to allocate 1073741815 entries for fdtab

### ISSUE THREAD

**Issue by adriangabura** [docker-library/haproxy#180]
I can't make 2.5 work on  my fedora 35 Workstation edition
```console
[NOTICE]   (1) : haproxy version is 2.5.5-384c5c5
[WARNING]  (1) : config : log format ignored for frontend 'http' since it has no log address.
[NOTICE]   (1) : New worker (11) forked
[NOTICE]   (1) : Loading success.
[WARNING]  (11) : [haproxy.main()] Failed to drop supplementary groups. Using 'gid'/'group' without 'uid'/'user' is generally useless.
[NOTICE]   (1) : haproxy version is 2.5.5-384c5c5
[ALERT]    (1) : Not enough memory to allocate 1073741815 entries for fdtab!
[ALERT]    (1) : No polling mechanism available.
  It is likely that haproxy was built with TARGET=generic and that FD_SETSIZE
  is too low on this platform to support maxconn and the number of listeners
  and servers. You should rebuild haproxy specifying your system using TARGET=
  in order to support other polling systems (poll, epoll, kqueue) or reduce the
  global maxconn setting to accommodate the system's limitation. For reference,
  FD_SETSIZE=1024 on this system, global.maxconn=536870898 resulting in a maximum of
  1073741815 file descriptors. You should thus reduce global.maxconn by 536870396. Also,
  check build settings using 'haproxy -vv'.
Thanks
```

**Comment by wglambert** [comment]
What's your `maxconn` set to?
https://git.pants-on.net/haproxy.git/commit/?h=haproxy-2.1/master&id=a0eeba891310ce85f2f348dd42c0fafec2e0e511

Using an example config with `maxconn 10000` I'm not able to reproduce. And when I use a value of `1000000` the logs show as
```console
[NOTICE]   (1) : haproxy version is 2.5.5-384c5c5
[NOTICE]   (1) : path to executable is /usr/local/sbin/haproxy
[ALERT]    (1) : [haproxy.main()] Cannot raise FD limit to 2000034, limit is 1048576.
```

**Comment by adriangabura** [comment]
I tried many values and nothing works. I have atm maxconn 40 and it fails. But if you downgrade to 2.4 or 2.3 it works with no such errors

**Comment by tianon** [comment]
Unfortunately, I don't think there's anything we can really do to fix this in the image -- for more help, I would suggest trying a dedicated support forum, such as [the Docker Community Forums](https://forums.docker.com/), [the Docker Community Slack](https://dockr.ly/slack), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker) (or maybe even something specific to haproxy instead).

**Comment by TimWolla** [comment]
> Unfortunately, I don't think there's anything we can really do to fix this in the image

I believe this issue can be solved with HAProxy 2.6 (to be released soon-ish) by leveraging the `fd-hard-limit` option in the configuration file: https://docs.haproxy.org/dev/configuration.html#fd-hard-limit
